devito/propagator.py

```diff
@@ -4,20 +4,48 @@
 from sympy import symbols, IndexedBase, Indexed
 from function_manager import FunctionDescriptor
 from collections import Iterable
+from at_controller import get_at_block_size, get_optimal_block_size
 
 
 class Propagator(object):
-    def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0, openmp=False, profile=False, cache_blocking=False, block_size=5):
+    def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0, openmp=False, profile=False,
+                 cache_blocking=False, block_size=None, auto_tune=False):
         self.t = symbols("t")
+
         self.cache_blocking = cache_blocking
-        if(isinstance(block_size, Iterable)):
-            if(len(block_size) == len(shape)):
-                self.block_sizes = block_size
+        self.auto_tune = auto_tune
+
+        self.spc_order = spc_border * 2  # setting space order
+        self.time_order = time_order
+
+        if self.cache_blocking:
```

Comment:
I do not agree with this because if auto_tuning was done (We only need to do it once as it takes ages) and report file was present:
We want to retrieve block sizes from that file (or using the equation @pvelesko  provided) or run auto-tuning again to update at-block sizes. Thus we are missing one flag.

Plus in my opinion its less confusing having one dedicated flag rather than playing with combinations of 2 vars. 

Moreover all the confusion can be easily avoided if proper documentation is written. Which I highly suggest doing with all the future classes etc.